source/guide_salt.rst

```diff
@@ -0,0 +1,153 @@
+.. author:: ctr49 <https://github.com/ctr49>
+
+.. tag:: automation
+
+.. highlight:: console
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/saltstack.png
+      :align: center
+
+##########
+SaltStack
+##########
+
+.. tag_list::
+
+`SaltStack`_ (or simply `Salt`) is Python-based, open-source software for event-driven IT automation, remote task execution, and configuration management. Supporting the "Infrastructure as Code" approach to data center system and network deployment and management, configuration automation, SecOps orchestration, vulnerability remediation, and hybrid cloud control.
+
+The ``salt-master`` controls so called ``salt-minions`` through formulae, pillars and grains. The scope of this guide is to install a ``salt-master``(without a co-located minion) on an Uberspace.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`Opening ports <basics-ports>`
+  * :manual:`supervisord <daemons-supervisord>`
+
+Prerequisites
+=============
+
+We need two open TCP ports for minions to communicate with the Salt master.
+
+.. code-block:: console
+
+ [isabell@stardust ~]$ uberspace port add
+ Port <high-port1> will be open for TCP and UDP traffic in a few minutes.
+ [isabell@stardust ~]$ uberspace port add
+ Port <high-port2> will be open for TCP and UDP traffic in a few minutes.
+ [isabell@stardust ~]$ uberspace port list
+ <high-port1>
+ <high-port2>
+
+Take note of the high ports, you will need them for salt master configuration later.
+
+Installation
+============
+
+Install
+-------
+
+Installing Salt in a ``virtualenv`` using python 3.8 (latest available python on Uberspace). Also installing pygit2 (because it is a dependency for any gitfs in Salt which is very common). Packages are installed in multiple steps as updated setuptools, pip and wheel may be needed for others to install. Salt must be installed by itself due to 'global-option' being used. Download latest sample config as last step.
+
+.. code-block:: console
+
+ [isabell@stardust ~]$ virtualenv -p python3.8 ~/salt_venv
```

Comment:
Do we really need a virtualenv here? Most other guides install using `pip3.8 install --user`.